controllers/testutils_test.go

```diff
@@ -29,31 +31,82 @@ const (
 )
 
 var (
-	defaultDuration = metav1.Duration{Duration: time.Minute * 5}
+	defaultDuration = metav1.Duration{Duration: time.Minute * 5}     // treat as const
+	letterRunes     = []rune("abcdefghijklmnopqrstuvwxyz1234567890") // treat as const
 )
 
-func defaultAppGroup(targetNamespace string) *v1alpha1.ApplicationGroup {
+func objKeyBuilder(name, namespace string) types.NamespacedName {
+	return types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}
+}
+
+func workflowBuilder(name, namespace string) *v1alpha13.Workflow {
```

Comment:
This also abstracts the functionality away for no reason in a test case. IMO, this makes it more difficult to read and understand what the test case is accomplishing

Reply:
Fixed